xds/internal/xdsclient/xdsresource/unmarshal_rds_test.go

```diff
@@ -166,6 +166,12 @@ func (s) TestRDSGenerateRDSUpdateFromRouteConfiguration(t *testing.T) {
 		defaultRetryBackoff = RetryBackoff{BaseInterval: 25 * time.Millisecond, MaxInterval: 250 * time.Millisecond}
 	)
 
+	oldRLS := envconfig.XDSRLS
```

Comment:
Please make sure there are tests that leave it disabled (or disable explicitly, either way), and don't produce errors.

Reply:
Hmm ok will add tests where it's disabled and it sees a cluster specifier plugin in a route, which it should NACK as not supported, and also ignore a top level Cluster Specifier Plugin list in the top level Route Configuration. I was debating between NACKING any length top level Route Configuration, or just ignoring it, but I think ignoring it is fine, as if there is no routes that specifiy to use a cluster specifier plugin the top level would be a no op anyway, otherwise it would get NACKED. I don't know if this is what you were going for?

Reply:
The goal is that it should work exactly the same as it did in 1.42 if the environment variable is not set.  So, it should completely ignore any cluster specifier plugin related config, except fail the route if a plugin is specified in an action since that's not something we would allow.

Reply:
Ah, cool, that's what my gut told me and that is the functionality in this pr. Keeping it exactly same as previously gives me a logical framework to think about environment variables objectively though, thanks.